website/docs/basics/autowiring.md

```diff
@@ -6,3 +6,213 @@ sidebar_label: Autowiring
+ * QueryDocuments class.
+ */
+class QueryDocuments implements ServiceInterface, QueryDocumentsInterface
+{
+    /**
+   * Get structured documents data for documents section.
+   *
+   * @param int $id Id for data.
+   *
+   * @return string
+   */
+  public function getDocumentsData(int $id = 0): string
+  {
+    ... here goes the implementation of getDocumentsData method.
+  }
+}
+```
+
+**src/Rest/Routes/DocumentsRoute.class**
+```js
+/**
+ * The class register route for DocumentsRoute endpoint
+ *
+ * @package ProjectNamespace\Rest\Routes
+ */
+
+declare(strict_types=1);
+
+namespace ProjectNamespace\Rest\Routes;
+
+use ProjectNamespace\Query\Documents\QueryDocumentsInterface;
+use ProjectNamespaceVendor\EightshiftLibs\Rest\Routes\AbstractRoute;
+use ProjectNamespaceVendor\EightshiftLibs\Rest\CallableRouteInterface;
+
+/**
+ * Class DocumentsRoute
+ */
+class DocumentsRoute extends AbstractRoute implements CallableRouteInterface
+{
+
+  /**
+   * Instance variable of documents data.
+   *
+   * @var QueryDocumentsInterface
+   */
+  protected $documentsData;
+
+  /**
+   * Create a new instance.
+   *
+   * @param QueryDocumentsInterface $queryDocuments Inject documentsData which holds data for documents.
+   */
+  public function __construct(QueryDocumentsInterface $queryDocuments)
+  {
+    $this->documentsData = $queryDocuments;
+  }
+
+  ...here goes the rest of the DocumentsRoute implementation.
+}
+```
+
+An this is it, autowiring knows how to resolve this setup, and you are all set.
+
+### What if I have to mock or do some weird stuff in my DI?
+
+Sometimes automatization can save you, but in the end, eventually, you need something custom, we got you covered.
+
+We can think of these scenarios:
+- You have to write tests and manually provide mocked classes.
+- You need to do some weird stuff in your DI.
+- Your classes have some custom structure, and autowiring can't resolve it.
+
+In those cases, you can manually provide your DI container with the implementation using the `getServiceClasses` method inside the `src>Main>Main` class.
+
+Just provide the method and add your custom implementation like this:
+
+`src>Main>Main.class`
+```js
+  /**
+   * Get the list of services to register.
+   *
+   * A list of classes which contain hooks.
+   *
+   * @return array<class-string, string|string[]> Array of fully qualified service class names.
+   */
+  protected function getServiceClasses(): array
+  {
+    return [
+
+      // If you are using a class as a DI.
+      ProjectNamespace\Rest\Routes\DocumentsRoute => [ProjectNamespace\Query\Documents\QueryDocuments],
+
+      // If you just want to include a simple class with no DI.
+      ProjectNamespace\CoolFolder\CoolClass.class,
+    ];
+  }
+```
+
+## Sum it up
+
+We understand that this is a long chapter but take your time and try to understand it because it can make you a better developer, your project better, and your clients happier. We tried to explain this as simply as possible.
```

Comment:
```suggestion
We understand that this is a long chapter but take your time and try to understand it because it can make you a better developer, your project better, and your clients happier. We tried to explain this as simply as possible.
```

Again, we should avoid making these claims. Using DIC doesn't automatically make you a better developer (oh how I wish it did 😄 ). Creating a good architecture and making your code testable will help you out in the long run (and that's the aim of all this). You can still use DIC and make shitty architecture that will bite you in the ass 😂 

I'd go with something along the lines of:

This chapter may be a bit long, but we wanted to make sure you understand our motivation behind it. We often work on large projects, with lots of third-party integrations, and having a good project structure and architecture makes the projects more durable. It makes them less prone to bugs (if you write tests), makes the implementation of new features, and removing obsolete features a lot easier.

Do you need to use it? Not really. Especially on really simple projects. Should you? We think you'll benefit from it in the long run.